scripts/install_third_party_libs_test.py

```diff
@@ -44,41 +45,122 @@
 class InstallThirdPartyLibsTests(test_utils.GenericTestBase):
     """Test the methods for installing third party libs."""
 
-    def setUp(self):
+    def setUp(self) -> None:
         super(InstallThirdPartyLibsTests, self).setUp()
 
         self.check_function_calls = {
             'check_call_is_called': False,
+            'extract_all_is_called': False,
+            'pre_push_hook_main_is_called': False,
+            'tweak_yarn_executable_is_called': False,
+            'mock_rename_called': False,
+            'url_retrieve_is_called': False,
+            'recursive_chmod_is_called': False
+        }
+        self.check_main_function_calls = {
+            'ensure_pip_library_is_installed_is_called': False,
+            'install_third_party_main_is_called': False,
+            'setup_main_is_called': False,
+            'setup_gae_main_is_called': False,
+            'pre_commit_hook_main_is_called': False
+        }
+        self.expected_check_main_function_calls = {
+            'ensure_pip_library_is_installed_is_called': True,
+            'install_third_party_main_is_called': True,
+            'setup_main_is_called': True,
+            'setup_gae_main_is_called': True,
+            'pre_commit_hook_main_is_called': True
         }
-        self.print_arr = []
-        def mock_check_call(unused_cmd_tokens, *args, **kwargs):  # pylint: disable=unused-argument
+        def mock_check_call(args: List[str], **kwargs: Union[int, None]) -> Ret:  # type: ignore[name-defined] # pylint: disable=unused-argument,undefined-variable
```

Comment:
Can we avoid using `# type: ignore[name-defined]`?

We could declare `Ret` outside and accept `returncode` and `communicate` message as well and write a common mock class that can be used here as well as for `mock_popen_error_call`. I think in that way we can avoid that ignore.

Reply:
Done.